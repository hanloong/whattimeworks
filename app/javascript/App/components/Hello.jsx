import React from 'react'
import ReactDOM from 'react-dom'

const Hello = ({ name }) => (
  <div>Hello {name}!</div>
)

Hello.defaultProps = {
  name: 'David'
}

Hello.propTypes = {
  name: React.PropTypes.string
}

module.exports = Hello;
