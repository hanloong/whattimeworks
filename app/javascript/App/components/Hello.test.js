import React from 'react';
import Hello from './Hello';
import renderer from 'react-test-renderer';

test('Hello prints out a hame', () => {
  const component = renderer.create(
    <Hello />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
