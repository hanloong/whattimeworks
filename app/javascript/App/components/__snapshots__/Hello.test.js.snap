// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello prints out a hame 1`] = `
<div>
  Hello 
  David
  !
</div>
`;
